risc0/zkp/build.rs

```diff
@@ -15,9 +15,11 @@
 use std::env;
 
 fn main() {
-    if env::var("CARGO_FEATURE_METAL").is_ok() {
-        let metal_bin = env::var("DEP_RISC0_SYS_METAL_KERNELS_ZKP")
-            .expect("CARGO_FEATURE_METAL is defined, but DEP_RISC0_SYS_METAL_KERNELS_ZKP is not");
+    if env::var("CARGO_FEATURE_PROVE").is_ok()
+        && env::var("CARGO_CFG_TARGET_OS").is_ok_and(|os| os == "macos" || os == "ios")
+    {
+        let metal_bin =
+            env::var("DEP_RISC0_SYS_METAL_KERNELS_ZKP").expect("DEP_RISC0_SYS_METAL_KERNELS_ZKP");
```

Comment:
Perhaps something like this:
```suggestion
            env::var("DEP_RISC0_SYS_METAL_KERNELS_ZKP").expect("attempting to build risc0-zkp using the `prove` feature flag. This requires building the metal kernels but DEP_RISC0_SYS_METAL_KERNELS_ZKP is not defined.");
```